File: 2015/day2/day2-part1.js
// Information of Puzzle Day2 - Part 1: https://adventofcode.com/2015/day/2

// Given a list of the dimensions (length(l), width(w), and hegiht(h)) of each present. (All numbers are in feets)
// The required wrapping paper for each present is: 2*(l*w + w*h + l*h) + Math.min(l*w, w*h, l*h)
// How many total square feet of wrapping paper should they order?

const fs = require("fs");

const getTotalWrappingPaper = function () {
  fs.readFile("day2-input.txt", (err, data) => {
    let total = 0;
    const file = data.toString();
    const lines = file.split(/\r?\n/);

    for (let line of lines) {
      dimensions = line.split("x");
      const l = parseInt(dimensions[0], 10);
      const w = parseInt(dimensions[1], 10);
      const h = parseInt(dimensions[2], 10);

      total += 2 * (l * w + w * h + l * h) + Math.min(l * w, w * h, l * h);
    }

    console.log("Total: ", total);
  });
};

getTotalWrappingPaper();

File: 2015/day2/day2-part1.py
# Information of Puzzle Day2 - Part 1: https://adventofcode.com/2015/day/2

# Given a list of the dimensions (length(l), width(w), and hegiht(h)) of each present. (All numbers are in feets)
# The required wrapping paper for each present is: 2*(l*w + w*h + l*h) + min(l*w, w*h, l*h)
# How many total square feet of wrapping paper should they order?

import sys


def get_total_wrapping_paper():
    with open('./day2-input.txt', 'r') as file:
        lines = file.read().splitlines()

    presents = (line.split('x') for line in lines)
    total = 0

    for present in presents:
        l = int(present[0])
        w = int(present[1])
        h = int(present[2])

        total += 2*(l*w + w*h + l*h) + min(l*w, w*h, l*h)

    print(total)

    return total


if __name__ == '__main__':
    sys.exit(get_total_wrapping_paper())
